src/mlpack/core/util/size_checks.hpp

```diff
@@ -31,16 +31,43 @@ template<typename DataType, typename LabelsType>
 inline void CheckSameSizes(const DataType& data,
                            const LabelsType& label,
                            const std::string& callerDescription,
-                           const std::string& addInfo = "labels")
-{
-  if (data.n_cols != label.n_cols)
+                           const std::string& addInfo = "labels",
+                           const std::string& isTranspose = "none")
```

Comment:
Honestly I would suggest adding two boolean options; one to indicate that the data matrix is transposed, and one to indicate that the labels are transposed.  This also ensures that a developer cannot call `CheckSameSizes()` with an invalid option.

Also, can you add some documentation for these new options?

Reply:
Sure ! Thats seems more appropriate. I will make required changes and add some documentation.